org.mixedrealitytoolkit.input/Tracking/TrackedPoseDriverWithFallback.cs

```diff
@@ -0,0 +1,168 @@
+
+        #endregion Fallback action values
+
+        #region TrackedPoseDriver Overrides 
+        /// <inheritdoc />
+        protected override void PerformUpdate()
+        {
+            base.PerformUpdate();
+
+            if (trackingStateInput.action == null)
+            {
+                Debug.LogWarning("TrackedPoseDriverWithFallback.trackingStateInput.action is null, no fallback will be used.");
+                return;
+            }
+
+            var hasPositionAction = positionAction != null;
+            var hasPositionFallbackAction = fallbackPositionAction != null;
+
+            var hasRotationAction = rotationAction != null;
+            var hasRotationFallbackAction = fallbackRotationAction != null;
+
+            InputTrackingState inputTrackingState = (InputTrackingState)trackingStateInput.action.ReadValue<int>();
+            InputTrackingState fallbackInputTrackingState = InputTrackingState.None;
+
+            // If default InputTrackingState does not have position and rotation data, use fallback if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && !inputTrackingState.HasFlag(InputTrackingState.Rotation) && FallbackTrackingStateAction.action != null)
+            {
+                inputTrackingState = (InputTrackingState)FallbackTrackingStateAction.action.ReadValue<int>();
+            }
+
+            if (FallbackTrackingStateAction.action != null)
+            {
+                fallbackInputTrackingState = (InputTrackingState)FallbackTrackingStateAction.action.ReadValue<int>();
+            }
+
+            bool neededToGetFallbackData = false;
+            Vector3 position = transform.localPosition;
+            Quaternion rotation = transform.localRotation;
+
+            // If no position data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && hasPositionFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                position = fallbackPositionAction.action.ReadValue<Vector3>();
+            }
+
+            // If no rotation data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Rotation) && hasRotationFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                rotation = fallbackRotationAction.action.ReadValue<Quaternion>();
+            }
+
+            if (neededToGetFallbackData) //because either position, rotation, or both data were obtained from fallback actions
+            {
+                SetLocalTransformFromFallback(position, rotation, (TDPwithFallbackTrackingStates)fallbackInputTrackingState);
+            }
+        }
+
+        protected virtual void SetLocalTransformFromFallback(Vector3 newPosition, Quaternion newRotation, TDPwithFallbackTrackingStates currentFallbackTrackingState)
```

Comment:
Make SetLocalTransformFromFallback private for now

Reply:
Done in https://github.com/ms-RistoRK/MixedRealityToolkit-Unity/commit/ad3addc04b06f24c1939dc02320f6517928dcfe5 , resolving.